src/vs/platform/accessibility/browser/accessibilityService.ts

```diff
@@ -30,20 +30,17 @@ export class AccessibilityService extends Disposable implements IAccessibilitySe
 	) {
 		super();
 		this._accessibilityModeEnabledContext = CONTEXT_ACCESSIBILITY_MODE_ENABLED.bindTo(this._contextKeyService);
-
-		const updateContextKey = () => this._accessibilityModeEnabledContext.set(this.isScreenReaderOptimized());
 		this._register(this._configurationService.onDidChangeConfiguration(e => {
 			if (e.affectsConfiguration('editor.accessibilitySupport')) {
-				updateContextKey();
+				this._updateContextKey();
 				this._onDidChangeScreenReaderOptimized.fire();
 			}
 			if (e.affectsConfiguration('workbench.reduceMotion')) {
 				this._configMotionReduced = this._configurationService.getValue('workbench.reduceMotion');
 				this._onDidChangeReducedMotion.fire();
 			}
 		}));
-		updateContextKey();
-		this._register(this.onDidChangeScreenReaderOptimized(() => updateContextKey()));
```

Comment:
we need to update the context key before `onDidChangeScreenReaderOptimized` because that gets listened to in the `editorStatus` and requires the updated value